### PR TITLE
170 us as a user i want to access transcendance from outside localhost

### PR DESCRIPTION
**Pull request by COTHSC**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [ ] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [x] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: nginx/certificates/transcendance.key [hunk 1]
```diff
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (llescure) [inline nginx/certificates/transcendance.key:1]: Isn't it a security failure to have our private key pushed in the repo ? Shouldn't it be in our .env ?

File: backend/app/src/proto/file.proto [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package userpackage;
+syntax = "proto3";
+
+mssage GameInfo {
```
Comment (llescure) [inline backend/app/src/proto/file.proto:4]: Rewrite mssage to message to make the server works

File: package.json [hunk 3]
```diff
@@ -22,23 +24,24 @@
...
+    "prisma:studio:test": "docker exec -it dev-backend npx dotenv -e .env.test -- prisma studio",
+    "prisma:migrate": "docker exec -it dev-backend npx prisma migrate dev",
+    "prisma:reset": "docker exec -it dev-backend npx prisma migrate reset -f",
     "dev:db:rm": "docker compose rm dev-db -s -f -v",
     "dev:db:up": "docker compose up dev-db -d",
     "db:dev:restart": "npm run dev:db:rm && npm run dev:db:up",
-    "backend:rm": "docker compose rm backend -s -f -v",
-    "backend:up": "docker compose up backend -d",
-    "backend:restart": "npm run backend:rm && npm run backend:up",
+    "backend:rm": "docker compose rm deb-backend -s -f -v",
```
Comment (llescure) [inline package.json:36]: Shouldn't it be dev-backend?

**Comment by llescure** [comment]
Can't send a message in channel
Screenshot 2022-11-18 at 12 48 17

**Review by llescure — CHANGES_REQUESTED** [review]
Can't send a message to channel 
Screenshot 2022-11-18 at 12 48 17

**Review by llescure — CHANGES_REQUESTED** [review]
Can't upload a new picture 
Screenshot_2022-11-18_at_12 42 45

**Review by llescure — CHANGES_REQUESTED** [review]
Can't get user achievement
Screenshot 2022-11-18 at 12 58 52

File: nginx/certificates/transcendance.key (hunk 1 above)
Comment (COTHSC) [reply nginx/certificates/transcendance.key:1]: Its a self signed certificate, so its not technically very bad to have it for development, but yes, when we get a proper certificate we should encrypt it with blackbox

File: backend/app/src/proto/file.proto (hunk 2 above)
Comment (COTHSC) [reply backend/app/src/proto/file.proto:4]: Oops, sorry about that typo just before committing, fixed!

File: package.json (hunk 3 above)
Comment (COTHSC) [reply package.json:36]: Yep that makes sense

**Comment by cnstll** [comment]
@COTHSC we should add all of the issues above as bugs no ? The PR looks big enough

**Comment by COTHSC** [comment]
> @COTHSC we should add all of the issues above as bugs no ? The PR looks big enough

No, all the above are caused by hardcoded "localhost" calls to the backend, the fix is trivial and will not significantly change the code

**Review by llescure — APPROVED** [review]
Looks good to me

**Pull request merged**